debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/SourceInfo.java

```diff
@@ -457,6 +477,47 @@ public void completeSnapshot() {
     }
 
     /**
+     * Set the filter data for the offset from the given {@link Filters}
+     * @param filters the filters
+     */
+    public void setFilterData(Filters filters) {
+        this.databaseWhitelist = filters.getDatabaseWhitelist();
+        this.databaseBlacklist = filters.getDatabaseBlacklist();
+        this.tableWhitelist = filters.getTableWhitelist();
+        this.tableBlacklist = filters.getTableBlacklist();
+    }
+
+    /**
+     * @return true if this offset has filter info, false otherwise.
+     */
+    public boolean hasFilterInfo() {
+        /**
+         * There are 2 possible cases for us not having filter info.
+         * 1. The connector does not use a filter. Creating a filter in such a connector could never add any tables.
+         * 2. The initial snapshot occurred in a version of Debezium that did not store the filter information in the
+         *    offsets.
+         */
+        return databaseWhitelist != null || databaseBlacklist != null ||
```

Comment:
What about the case where we e.g. have a table whitelist but not database whitelist? It seems we'd re-set it in the `SourceInfo` unless _all_ values had been present before?

Reply:
If at least one whitelist/blacklist is not null, then I want this method to return true, because then there is (some amount of) filter info. That seems to be what it is doing?